cmake/install.cmake

```diff
@@ -70,41 +75,36 @@ SET(INSTALL_DOCDIR_DEFAULT "docs")
 SET(INSTALL_LICENSEDIR_DEFAULT ${INSTALL_DOCDIR_DEFAULT})
 IF(NOT IS_SUBPROJECT)
   SET(INSTALL_PLUGINDIR_DEFAULT "lib/mariadb/plugin")
-ELSE()
 ENDIF()
 SET(LIBMARIADB_STATIC_DEFAULT "mariadbclient")
+
 #
 # RPM layout
 #
-SET(INSTALL_BINDIR_RPM "bin")
-IF((CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le" OR CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "s390x") AND CMAKE_SIZEOF_VOID_P EQUAL 8)
-  SET(INSTALL_LIBDIR_RPM "lib64/mariadb")
-  SET(INSTALL_PCDIR_RPM "lib64/pkgconfig")
-  SET(INSTALL_PLUGINDIR_RPM "lib64/mariadb/plugin")
-ELSE()
-  SET(INSTALL_LIBDIR_RPM "lib/mariadb")
-  SET(INSTALL_PCDIR_RPM "lib/pkgconfig")
-  SET(INSTALL_PLUGINDIR_RPM "lib/mariadb/plugin")
-ENDIF()
-SET(INSTALL_INCLUDEDIR_RPM "include")
-SET(INSTALL_DOCDIR_RPM "docs")
+SET(INSTALL_BINDIR_RPM "${CMAKE_INSTALL_BINDIR}")
+SET(INSTALL_LIBDIR_RPM "${CMAKE_INSTALL_LIBDIR}/mariadb")
+SET(INSTALL_PCDIR_RPM "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+SET(INSTALL_PLUGINDIR_RPM "${CMAKE_INSTALL_LIBDIR}/mariadb/plugin")
+SET(INSTALL_INCLUDEDIR_RPM "${CMAKE_INSTALL_INCLUDEDIR}/mariadb")
+SET(INSTALL_DOCDIR_RPM "${CMAKE_INSTALL_DOCDIR}")
+SET(INSTALL_LICENSEDIR_RPM "${INSTALL_DOCDIR_RPM}")
 SET(LIBMARIADB_STATIC_RPM "mariadbclient")
 
 #
 # DEB layout
 #
-SET(INSTALL_BINDIR_DEB "bin")
-SET(INSTALL_LIBDIR_DEB "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
-SET(INSTALL_PCDIR_DEB "lib/${CMAKE_LIBRARY_ARCHITECTURE}/pkgconfig")
+SET(INSTALL_BINDIR_DEB "${CMAKE_INSTALL_BINDIR}")
+SET(INSTALL_LIBDIR_DEB "${CMAKE_INSTALL_LIBDIR}/${CMAKE_LIBRARY_ARCHITECTURE}")
```

Comment:
I thought the whole point of using `GNUInstallDirs` was to avoid this debian specific architecture in the path. `GNUInstallDirs` documentation says

> ``LIBDIR``
>  object code libraries (``lib`` or ``lib64``
>  or ``lib/<multiarch-tuple>`` on Debian)

so *presumably* it's taken care of automatically.

---
On the other hand, this is an argument *against* `GNUInstallDirs`. It behaves differently on Debian, looking, exactly, at what system *you build on*. Currently I can build a set or RPM packages anywhere, I do it regularly on Gentoo, there's no need to have an rpm-based system for that. With `GNUInstallDirs` you can only build debian packages on Debian. Given that, and that you don't use `GNUInstallDirs` correctly anyway (it's included too early to set correct paths), I'd suggest not to use it at all.

Reply:
good point, thanks

Reply:
no, LIBDIR on debians is "lib". multiarch dir is handled separately. thinking about dropping GNUInstallDirs

Reply:
it's not just LIBDIR. it's automatic DOCDIR and more. I vote for GNUInstallDirs with maybe moderate usage